tripal/tripal
# GFF3 Importer: duplicate landmark uniquename bug

# Bug Fix

## Issue #1324 

## Description

The issue is that the landmark lookup for the GFF3 loader does not use all three unique constraints of the feature table, only ```uniquename``` and ```organism_id``` are used. Two features may share these and only differ by ```type_id```.

The first commit https://github.com/tripal/tripal/pull/1369/commits/895612170a230c3ba84ee7ddca61f401516b0e2f just adds an exception when we encounter a duplicate landmark uniquename for a given organism. This fails testing because ~~several tests have duplicates~~ it was implemented incorrectly.

## Testing?
1, 2. Load the same fasta file twice using the same organism but a different cv term each time, _e.g._ ```DNA``` and ```chromosome```. This will generate the duplicate landmark uniquenames.
3. Load a gff3 file, do not specify the "Landmark Type". On this branch it should fail with an error message like
```The landmark "XYZ_Chr1" is ambiguous for this organism. Please redo the importer job and specify a landmark type.```
4. Run it again specifying one of the two landmark types that you used. The file should load successfully.
